geode-core/src/test/java/org/apache/geode/internal/cache/DistributedCacheOperationTest.java

```diff
@@ -17,22 +17,68 @@
 import static org.assertj.core.api.Assertions.assertThat;
 import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.assertTrue;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.doThrow;
 import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.spy;
 import static org.mockito.Mockito.times;
 import static org.mockito.Mockito.verify;
 import static org.mockito.Mockito.when;
 
 import java.util.HashMap;
 import java.util.Map;
 
+import org.junit.Before;
 import org.junit.Test;
+import org.mockito.ArgumentCaptor;
 
+import org.apache.geode.cache.CacheClosedException;
 import org.apache.geode.cache.CacheEvent;
+import org.apache.geode.cache.EntryNotFoundException;
+import org.apache.geode.cache.Scope;
+import org.apache.geode.distributed.internal.ClusterDistributionManager;
+import org.apache.geode.distributed.internal.InternalDistributedSystem;
+import org.apache.geode.distributed.internal.OperationExecutors;
+import org.apache.geode.distributed.internal.ReplyException;
 import org.apache.geode.distributed.internal.membership.InternalDistributedMember;
 import org.apache.geode.internal.cache.DistributedCacheOperation.CacheOperationMessage;
 import org.apache.geode.internal.cache.persistence.PersistentMemberID;
+import org.apache.geode.internal.cache.versions.VersionTag;
+import org.apache.geode.test.fake.Fakes;
 
 public class DistributedCacheOperationTest {
+  private TestCacheOperationMessage message;
+  private InternalDistributedMember sender;
+  private ClusterDistributionManager dm;
+  private LocalRegion region;
+  private VersionTag<?> versionTag;
+  private Scope scope;
+  private OperationExecutors executors;
+  private final int processorId = 1;
+
+  @Before
+  public void setup() {
+    message = spy(new TestCacheOperationMessage());
+    sender = mock(InternalDistributedMember.class);
+    versionTag = mock(VersionTag.class);
+
+    GemFireCacheImpl cache = Fakes.cache();
```

Comment:
I'd really like to see us stop using `Fakes.cache()` in new unit tests so we can get rid of it. Can you please see if you can remove this and mock everything `DistributedCacheOperation` needs directly? Also, please try to use `InternalCache` and `InternalRegion` instead of `GemFireCacheImpl` and `LocalRegion`. Let me know if you'd like some help or want me to pair with you.

Reply:
Removed the Fakes.
As the product code method signature is using LocalRegion, using InternalRegion may need  some casting. Will leave it as is, unless we need to change the product code.